### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by dragonmaus**
This one is a bit more extensive in terms of lines changed, but none of the changes should affect functionality in any way.

Overall I tried to bring the formatting more in line with the standards you were already using, updated some deprecated constructs, and cleaned up some of the other minor issues I could find.

### CODE REVIEW

**Comment by Sollace** [comment]
You don't have to worry about conflicts. I can easily resolve those.

Style-wise, there are a few things I'm going to need changed. I'll leave some reviews for those.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# https://editorconfig.org/
```
Comment (Sollace) [inline .editorconfig]: Please remove this file

File: build.gradle [hunk 2]
```diff
@@ -1,3 +1,5 @@
+//file:noinspection GroovyUnusedAssignment
```
Comment (Sollace) [inline build.gradle]: No idea what this is

File: gradle.properties [hunk 3]
```diff
@@ -1,17 +1,18 @@
+# suppress inspection "UnusedProperty" for whole file
```
Comment (Sollace) [inline gradle.properties]: Please revert these indentation changes. This was done for grouping purposes.

File: src/main/java/com/sollace/coppergolem/CopperButtonBlock.java [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package com.sollace.coppergolem;
 
+import com.sollace.coppergolem.entity.CopperGolemEntity;
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/CopperButtonBlock.java]: Import order and groupings need to be preserved. I see a few places where they've been swapped around.

File: src/main/java/com/sollace/coppergolem/CopperButtonBlock.java [hunk 5]
```diff
@@ -32,7 +31,7 @@ public void onStateReplaced(BlockState state, World world, BlockPos pos, BlockSt
         super.onStateReplaced(state, world, pos, newState, moved);
 
         if (!moved && newState.getBlock() instanceof CopperButtonBlock && newState.get(POWERED)) {
-            world.createAndScheduleBlockTick(pos, newState.getBlock(), ((CopperButtonBlock)newState.getBlock()).getCustomPressTicks());
+            world.createAndScheduleBlockTick(pos, newState.getBlock(), ((CopperButtonBlock) newState.getBlock()).getCustomPressTicks());
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/CopperButtonBlock.java]: No spaces after casting, please.

File: src/main/java/com/sollace/coppergolem/GBlocks.java [hunk 6]
```diff
@@ -11,33 +12,31 @@
 import net.minecraft.util.Identifier;
 import net.minecraft.util.registry.Registry;
 
-import com.sollace.coppergolem.registry.MemoizeRegistries;
-
 public interface GBlocks {
     Block OXIDIZED_COPPER_BUTTON = register("oxidized_copper_button",
-            new OxidizableCopperButtonBlock(Oxidizable.OxidationLevel.OXIDIZED, AbstractBlock.Settings.of(Material.METAL, MapColor.TEAL).requiresTool().strength(3, 6).sounds(BlockSoundGroup.COPPER))
+        new OxidizableCopperButtonBlock(Oxidizable.OxidationLevel.OXIDIZED, AbstractBlock.Settings.of(Material.METAL, MapColor.TEAL).requiresTool().strength(3, 6).sounds(BlockSoundGroup.COPPER))
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/GBlocks.java]: Wrapped method parameters and wrapped method chains need to be double-indented.

File: src/main/java/com/sollace/coppergolem/GItems.java [hunk 7]
```diff
@@ -24,9 +24,10 @@ static Item register(String name, Block block) {
         return Registry.register(Registry.ITEM, new Identifier("copper_golem", name), new BlockItem(block, new Item.Settings().group(ItemGroup.REDSTONE)));
     }
 
-    static void bootstrap() {}
+    static void bootstrap() {
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/GItems.java]: One line, please.

File: src/main/java/com/sollace/coppergolem/client/CopperGolemEntityModel.java [hunk 8]
```diff
@@ -36,32 +35,32 @@ public CopperGolemEntityModel(ModelPart root) {
...
-                        .uv(36, 31).cuboid(-1, -1, -1.5F, 2, 2, 2)
-                        .uv(0, 0).cuboid(-4, -5, -4, 8, 5, 7)
-                        .uv(26, 16).cuboid(-1, -7, -1.5F, 2, 2, 2)
-                        .uv(0, 32).cuboid(-1.5F, -10, -2, 3, 3, 3), ModelTransform.pivot(0, -6, 0))
-                .addChild(EntityModelPartNames.NOSE, ModelPartBuilder.create()
-                        .uv(36, 26).cuboid(-1.5F, -1, 0, 2, 3, 2), ModelTransform.pivot(0.5F, -1, 3));
+                .uv(36, 31).cuboid(-1, -1, -1.5F, 2, 2, 2)
+                .uv(0, 0).cuboid(-4, -5, -4, 8, 5, 7)
+                .uv(26, 16).cuboid(-1, -7, -1.5F, 2, 2, 2)
+                .uv(0, 32).cuboid(-1.5F, -10, -2, 3, 3, 3), ModelTransform.pivot(0, -6, 0))
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/client/CopperGolemEntityModel.java]: See previous note about wrapped method chains. Some of these I've tried to line them up with the start of the first method in the chain as well.

File: src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java [hunk 9]
```diff
@@ -120,9 +118,7 @@ protected void initDataTracker() {
     }
 
     public BlockInteraction getFinder(int maxDistance) {
-        return finders.computeIfAbsent(Registry.ITEM.getId(getStackInHand(Hand.MAIN_HAND).getItem()), id -> {
-            return BlockInteraction.create(this, maxDistance);
-        });
+        return finders.computeIfAbsent(Registry.ITEM.getId(getStackInHand(Hand.MAIN_HAND).getItem()), id -> BlockInteraction.create(this, maxDistance));
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java]: Lambda formatting should not be changed from what they were before. I've used them with blocks in some places for readability purposes as otherwise the lines get too long to read.

File: src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java [hunk 10]
```diff
@@ -502,13 +497,14 @@ public float getSoundPitch() {
...
 
+    @SuppressWarnings("UnusedReturnValue")
     public static boolean tryBuild(World world, BlockPos pos) {
         BlockPattern.Result result = PATTERN.searchAround(world, pos);
         if (result == null) {
             return false;
         }
 
-        OxidationLevel[] oxidation = new OxidationLevel[] { OxidationLevel.UNAFFECTED };
+        OxidationLevel[] oxidation = new OxidationLevel[]{OxidationLevel.UNAFFECTED};
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java]: Array declarations must be spaced out.

File: src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java [hunk 11]
```diff
@@ -54,7 +53,7 @@ public Optional<BlockPos> pickAny(Predicate<BlockPos> filter) {
     }
 
     private Stream<BlockPos> findPositions() {
-        knownPositions.removeIf(p -> !entity.getEntityWorld().isChunkLoaded(p) || !isValid(toState(p)));
+        knownPositions.removeIf(p -> !entity.getEntityWorld().isChunkLoaded(p.getX(), p.getZ()) || !isValid(toState(p)));
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java:57]: What is the purpose of this change?

File: src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java [hunk 12]
```diff
@@ -69,8 +68,8 @@ private Stream<BlockPos> findPositions() {
 
     protected Stream<BlockPos> searchArea(int range) {
         var stream = BlockPos.streamOutwards(entity.getBlockPos(), range, range, range)
-                .filter(this::isExposed)
-                .filter(p -> entity.getEntityWorld().isChunkLoaded(p) && isValid(toState(p)));
+            .filter(this::isExposed)
+            .filter(p -> entity.getEntityWorld().isChunkLoaded(p.getX(), p.getZ()) && isValid(toState(p)));
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java]: Same as above. If there's a single-parameter version of this method, please use that.

File: src/main/java/com/sollace/coppergolem/mixin/MixinMobEntity.java [hunk 13]
```diff
@@ -1,23 +1,23 @@
...
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
-import com.sollace.coppergolem.registry.MobEntityInitGoalsListener;
-
 @Mixin(MobEntity.class)
 abstract class MixinMobEntity extends LivingEntity {
-    MixinMobEntity() {super(null, null);}
+    MixinMobEntity() {
+        super(null, null);
+    }
```
Comment (Sollace) [inline src/main/java/com/sollace/coppergolem/mixin/MixinMobEntity.java]: One line please.

File: src/main/resources/assets/copper_golem/lang/en_us.json [hunk 14]
```diff
@@ -1,20 +1,20 @@
...
-   "subtitles.entity.copper_golem.no": "Copper Golem harrumphs",
-   "subtitles.entity.copper_golem.death": "Copper Golem dies",
-   "subtitles.entity.copper_golem.whirl": "Gear whirls"
+  "subtitles.entity.copper_golem.step": "Footsteps",
+  "subtitles.entity.copper_golem.ambient": "Copper Golem grunts",
+  "subtitles.entity.copper_golem.hurt": "Copper Golem hurts",
+  "subtitles.entity.copper_golem.no": "Copper Golem harrumphs",
+  "subtitles.entity.copper_golem.death": "Copper Golem dies",
+  "subtitles.entity.copper_golem.whirl": "Gear whirls"
 }
```
Comment (Sollace) [inline src/main/resources/assets/copper_golem/lang/en_us.json]: Json files should preferably keep their terminating newline. Some might be missing them, so should be updated add them.

**Review by Sollace — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (dragonmaus) [reply build.gradle]: These annotations are to keep IDEs and other tooling from complaining about unused variables/properties/etc., but have no impact on compilation or output and can be easily removed.

File: src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java (hunk 11 above)
Comment (dragonmaus) [reply src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java:57]: I made this change (and the one below) because `.isChunkLoaded(p)` is marked as deprecated; apparently Fabric (or Mojang) are removing it in favour of the two-argument version?

File: src/main/java/com/sollace/coppergolem/CopperButtonBlock.java (hunk 4 above)
Comment (dragonmaus) [reply src/main/java/com/sollace/coppergolem/CopperButtonBlock.java]: Done. I did notice that your `Main.java` file has the `com.sollace` group before the `net.fabricmc`/`net.minecraft` group, while it's after in all of the other files. Not sure if that matters, so I left it that way.

File: src/main/java/com/sollace/coppergolem/client/CopperGolemEntityRenderer.java [hunk 15]
```diff
@@ -1,16 +1,16 @@
...
+import net.minecraft.client.render.entity.MobEntityRenderer;
 import net.minecraft.client.render.entity.feature.FeatureRenderer;
 import net.minecraft.client.render.entity.feature.FeatureRendererContext;
 import net.minecraft.client.render.model.json.ModelTransformation;
 import net.minecraft.client.util.math.MatrixStack;
 import net.minecraft.entity.EquipmentSlot;
 import net.minecraft.item.ItemStack;
-import net.minecraft.client.MinecraftClient;
-import net.minecraft.client.render.VertexConsumerProvider;
-import net.minecraft.client.render.entity.MobEntityRenderer;
```
Comment (dragonmaus) [inline src/main/java/com/sollace/coppergolem/client/CopperGolemEntityRenderer.java:13]: I wasn't sure if this group being out-of-order matters, but it's easy enough for me to revert if it does.

File: src/main/java/com/sollace/coppergolem/registry/MemoizeBackedRegistry.java [hunk 16]
```diff
@@ -4,6 +4,7 @@
 
 import com.google.common.collect.BiMap;
 import com.google.common.collect.HashBiMap;
+
```
Comment (dragonmaus) [inline src/main/java/com/sollace/coppergolem/registry/MemoizeBackedRegistry.java:7]: Added this for consistency with the other files.

File: gradle.properties [hunk 17]
```diff
@@ -9,9 +9,9 @@ org.gradle.daemon=false
   fabric_version=0.43.1+1.18
 
 # Mod Properties
-	group=com.sollace
-	displayname=Copper Golem
-	authors=Sollace
+    group=com.sollace
+    displayname=Copper Golem
+    authors=Sollace
```
Comment (dragonmaus) [inline gradle.properties:14]: This is a "tab to space" change; let me know if that matters.

File: src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java [hunk 18]
```diff
@@ -51,8 +51,8 @@
 import com.sollace.coppergolem.GItems;
 import com.sollace.coppergolem.GSounds;
 import com.sollace.coppergolem.entity.ai.BlockInteraction;
-import com.sollace.coppergolem.entity.ai.PressButtonGoal;
 import com.sollace.coppergolem.entity.ai.ChaseTargetGoal;
+import com.sollace.coppergolem.entity.ai.PressButtonGoal;
```
Comment (dragonmaus) [inline src/main/java/com/sollace/coppergolem/entity/CopperGolemEntity.java:55]: Ditto for the comment about out-of-order imports in `client/CopperGolemEntityRenderer.java`.

File: src/main/resources/copper_golem.mixin.json [hunk 19]
```diff
@@ -1,13 +1,12 @@
...
+  "required": true,
+  "minVersion": "0.7",
+  "package": "com.sollace.coppergolem.mixin",
+  "refmap": "copper_golem.mixin.refmap.json",
+  "compatibilityLevel": "JAVA_16",
+  "mixins": [
+    "MixinAbstractButtonBlock",
+    "MixinMobEntity"
+  ],
+  "client": []
```
Comment (dragonmaus) [inline src/main/resources/copper_golem.mixin.json:11]: `jq` likes to compact empty arrays. Let me know if you prefer the two-line version.

**Comment by dragonmaus** [comment]
That should cover most everything you commented on. I left comments where I was uncertain, let me know what you think.

File: src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java (hunk 11 above)
Comment (Sollace) [reply src/main/java/com/sollace/coppergolem/entity/ai/PositionFinder.java:57]: The deprecation annotations come from Mojang, and the way they use it is... A little weird. It doesn't mean they're going to remove, but just that it's not meant to be called directly.

It's like the methods in Block and Item, where deprecated just means overriding/implementing is fine, but calling usually goes through BlockState/ItemStack. It's not a big deal, so I guess this change can stay. Just remember that for other changes moving forward.

File: gradle.properties (hunk 17 above)
Comment (Sollace) [reply gradle.properties:14]: Should be fine. Thanks!

File: src/main/resources/copper_golem.mixin.json (hunk 19 above)
Comment (Sollace) [reply src/main/resources/copper_golem.mixin.json:11]: Single-line empty arrays is preferrable. Thanks!

File: src/main/java/com/sollace/coppergolem/CopperButtonBlock.java (hunk 4 above)
Comment (Sollace) [reply src/main/java/com/sollace/coppergolem/CopperButtonBlock.java]: You can fix that if you want, otherwise everything seems okay.

File: build.gradle (hunk 2 above)
Comment (Sollace) [reply build.gradle]: imo I'd just turn off the inspection in IDEA's settings, but I suppose this is fine, since it doesn't affect anything.

**Review by Sollace — APPROVED** [review]
